.circleci: Fold postnightly workfow into nightly

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#39669 .circleci: Fold postnightly workfow into nightly**

Folds the postnightly workflow, including html updating jobs and binary
size jobs, into the regular nightly workflow that should only run after
all upload jobs have completed.

This also moves the smoke testing jobs into the binary_builds workflow.

Do note that the devtoolset7 html update job has been removed since we
do not upload binaries specifically to that location anymore.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D21936811](https://our.internmc.facebook.com/intern/diff/D21936811)

seemethere committed:
.circleci: Fold postnightly workfow into nightly

Folds the postnightly workflow, including html updating jobs and binary
size jobs, into the regular nightly workflow that should only run after
all upload jobs have completed.

This also moves the smoke testing jobs into the binary_builds workflow.

Do note that the devtoolset7 html update job has been removed since we
do not upload binaries specifically to that location anymore.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Fold postnightly workfow into nightly"


Folds the postnightly workflow, including html updating jobs and binary
size jobs, into the regular nightly workflow that should only run after
all upload jobs have completed.

This also moves the smoke testing jobs into the binary_builds workflow.

Do note that the devtoolset7 html update job has been removed since we
do not upload binaries specifically to that location anymore.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

malfet commented on .circleci/cimodel/data/binary_build_definitions.py:
```diff
@@ -116,6 +105,19 @@ def gen_workflow_job(self, phase, upload_phase_dependency=None, nightly=False):
         job_name = "_".join([self.get_name_prefix(), os_name, phase])
         return {job_name : job_def}
 
+def gen_binary_filter_branch(branch):
```
Can you extend https://github.com/pytorch/pytorch/blob/df2d19723a64304aa0ca5b6887c649e25b5c2751/.circleci/cimodel/data/simple/util/branch_filters.py#L8
(by adding optional `tags` member) rather than create another inline copy here?

seemethere replied:
done

seemethere committed:
Update on ".circleci: Fold postnightly workfow into nightly"


Folds the postnightly workflow, including html updating jobs and binary
size jobs, into the regular nightly workflow that should only run after
all upload jobs have completed.

This also moves the smoke testing jobs into the binary_builds workflow.

Do note that the devtoolset7 html update job has been removed since we
do not upload binaries specifically to that location anymore.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D21936811](https://our.internmc.facebook.com/intern/diff/D21936811)

[ghstack-poisoned]
